File: opensrp-chw-referral/src/main/java/org/smartregister/chw/referral/domain/ReferralTask.java
package org.smartregister.chw.referral.domain;

import org.smartregister.clientandeventmodel.Event;

public class ReferralTask {
    private String groupId;
    private String referralDescription;
    private String focus;
    private Event event;

    public String getGroupId() {
        return groupId;
    }

    public void setGroupId(String groupId) {
        this.groupId = groupId;
    }

    public String getReferralDescription() {
        return referralDescription;
    }

    public void setReferralDescription(String referralDescription) {
        this.referralDescription = referralDescription;
    }

    public String getFocus() {
        return focus;
    }

    public void setFocus(String focus) {
        this.focus = focus;
    }

    public Event getEvent() {
        return event;
    }

    public void setEvent(Event event) {
        this.event = event;
    }
}

File: opensrp-chw-referral/src/main/java/org/smartregister/chw/referral/domain/ReferralTask.kt
package org.smartregister.chw.referral.domain

import org.smartregister.clientandeventmodel.Event

/**
 * This is the wrapper class for both Event and Task. When a client is referred, a referral [Event] is
 * created from the provided [event] then a FHIR based task is also generated and synced to OpenSRP
 * app server. The task specification details can be found here:
 * @see [OpenSRP Referral Task Specification](https://smartregister.atlassian.net/wiki/spaces/Documentation/pages/1156251738/FHIR+Task+Based+Referrals+Software+Requirements+Specification)
 */
data class ReferralTask(var event: Event) {
    var groupId: String? = null
    var referralDescription: String? = null
    var focus: String? = null
}